source/posix/rw_lock.c

```diff
@@ -12,48 +12,103 @@
  * express or implied. See the License for the specific language governing
```

Comment:
_Please_ document the protocol being implemented for this rwlock. It's non-obvious what is going on here.